plugins/inputs/x509_cert/x509_cert.go

```diff
@@ -129,10 +139,15 @@ func getFields(cert *x509.Certificate, now time.Time) map[string]interface{} {
 	return fields
 }
 
-func getTags(subject pkix.Name, location string) map[string]string {
+func (c *X509Cert) getTags(cert *x509.Certificate, location string) map[string]string {
+	subject := cert.Subject
+
 	tags := map[string]string{
-		"source":      location,
-		"common_name": subject.CommonName,
+		"source":               location,
+		"common_name":          subject.CommonName,
+		"serial_number":        cert.SerialNumber.Text(16),
```

Comment:
The certificate serial number is at most 20 bytes and this will be at most 40 Hex characters. This is supposed to be unique per CA. 

On the other hand, based on what I know, `cert.Subject.SerialNumber` is an optional serial number that the public key owner can include. It's probably not as useful to disambiguate certificates than the certificate serial.

Reply:
I have added a new test.